.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,111 @@
+name: Pull Request CI
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" or "develop" branch
+  pull_request:
+    branches:
+      - main
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+    paths-ignore:
+      - 'infra-l2-dynamo/**'
+      - 'infra-l2-kms/**'
+      - 'infra-l2-waf/**'
+      
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Required for sam-validate step
+permissions:
+  id-token: write
+  contents: read
+
+env: # Only adding the variables in that are required for
+  AWS_REGION: eu-west-2
+
+jobs:
+  run-code-check:
+    name: Code Checks
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: ./src/
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Checks if commiting secrets to repo
+        uses: pre-commit/action@v3.0.0
+        with:
+          extra_args: "detect-secrets > .secrets.baseline"
+
+      - name: Setup nodeJS v18
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Set offline mirror path
+        id: offline-mirror-path
+        run: echo "mirror-dir=${GITHUB_WORKSPACE}/npm-packages-offline-cache" >> $GITHUB_OUTPUT
+
+      - name: Cache npm offline-mirror
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.offline-mirror-path.outputs.mirror-dir }}
+          key: offline-mirror-oauth
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run lint
+        run: npm run lint
+
+      # - name: Run Tests
```

Comment:
need to uncomment them as we code to test now @Stelios-CB

Reply:
[Resolved](https://github.com/alphagov/di-ipv-cri-f2f-api/pull/9/commits/68cddb6d5153c3ca17eb65bc33614ac1cc2653bf)